lib/monitoring/metrics/policy_resouce_gauge.rb

```diff
@@ -0,0 +1,29 @@
+module Monitoring
+  module Metrics
+    class PolicyResourceGauge
+      attr_reader :registry, :pubsub, :metric_name, :docstring, :labels, :sub_event_name
+
+      def setup(registry, pubsub)
+        @registry = registry
+        @pubsub = pubsub
+        @metric_name = :conjur_resource_count
+        @docstring = 'Number of resources in Conjur database'
+        @labels = %i[kind]
+        @sub_event_name = 'conjur.policy_loaded'
+
+        # Create/register the metric
+        Metrics.create_metric(self, :gauge)
+        # Run update to set the initial counts
+        update
+      end
+
+      def update(*payload)
+        metric = @registry.get(@metric_name)
+        kind = ::Sequel.function(:kind, :resource_id)
+        Resource.group_and_count(kind).each do |record|
+          metric.set(record[:count], labels: { kind: record[:kind] })
+        end
+      end
```

Comment:
I think what you're suggesting would be termed "debouncing". This might result in a metric update only taking place after an extended of time after the policy load activity has calmed down. 

Instead I think we want to make sure that there is a limited number of metric updates within some repeating time interval. I think this behaviour is termed "throttling". This link helped me distinguish between the two  https://web.archive.org/web/20180324022838/http://demo.nimius.net/debounce_throttle/.

The idea is some like this

1. We start with no timer
1. When a DB query is triggered we must start an X (e.g. X=5) second timer.
1. When there is no active timer. If a policy load event is triggered then immediately trigger a DB query.
1. When there is an active timer. If a policy load event is triggered then when the timer is exhausted we trigger a DB query.
1. When there is an active timer. If no policy load event is triggered then remove the timer when it is exhausted.